src/java.base/share/classes/java/lang/StableValue.java

```diff
@@ -0,0 +1,664 @@
+ *           {@linkplain #list(int, IntFunction) a stable list} of arbitrary depth, which
+ *           provides stable components. More generally, a stable value can hold other
+ *           stable values of arbitrary depth and still provide transitive constantness.
+ *
+ * @param <T> type of the content
+ *
+ * @since 25
+ */
+@PreviewFeature(feature = PreviewFeature.Feature.STABLE_VALUES)
+public sealed interface StableValue<T>
+        permits StableValueImpl {
+
+    // Principal methods
+
+    /**
+     * {@return {@code true} if the content was set to the provided {@code value},
+     *          {@code false} otherwise}
+     * <p>
+     * When this method returns, the content is always set.
+     *
+     * @param value to set
+     */
+    boolean trySet(T value);
+
+    /**
+     * {@return the content if set, otherwise, returns the provided {@code other} value}
+     *
+     * @param other to return if the content is not set
+     */
+    T orElse(T other);
+
+    /**
+     * {@return the content if set, otherwise, throws {@code NoSuchElementException}}
+     *
+     * @throws NoSuchElementException if no content is set
+     */
+    T orElseThrow();
+
+    /**
+     * {@return {@code true} if the content is set, {@code false} otherwise}
+     */
+    boolean isSet();
+
+    /**
+     * {@return the content; if unset, first attempts to compute and set the
+     *          content using the provided {@code supplier}}
+     * <p>
+     * The provided {@code supplier} is guaranteed to be invoked at most once if it
+     * completes without throwing an exception.
+     * <p>
+     * If the supplier throws an (unchecked) exception, the exception is rethrown, and no
+     * content is set. The most common usage is to construct a new object serving
+     * as a lazily computed value or memoized result, as in:
+     *
+     * {@snippet lang=java:
+     * Value witness = stable.orElseSet(Value::new);
+     * }
+     * <p>
+     * When this method returns successfully, the content is always set.
+     * <p>
+     * This method will always return the same witness value regardless if invoked by
+     * several threads. Also, the provided {@code supplier} will only be invoked once even
+     * if invoked from several threads unless the {@code supplier} throws an exception.
+     *
+     * @param  supplier to be used for computing the content, if not previously set
+     * @throws IllegalStateException if the provided {@code supplier} recursively
+     *                               attempts to set this stable value.
+     */
+    T orElseSet(Supplier<? extends T> supplier);
+
+    // Convenience methods
+
+    /**
+     * Sets the content to the provided {@code value}, or, if already set,
+     * throws {@code IllegalStateException}.
+     * <p>
+     * When this method returns (or throws an exception), the content is always set.
+     *
+     * @param value to set
+     * @throws IllegalStateException if the content was already set
```

Comment:
The javadoc speaks about "a stable value is set", and not "content is set". Make sure we're using terminology consistenty